### PR TITLE
More auditing

### PR DESCRIPTION
**Pull request by SmittieC**
Addresses #144

For a smooth experience, go through the PR commit-wise.

All went well, except for the `Membership`model. I had to update the tests, since `team.members.add(...)` does not play well with the auditing library, which is a suboptimality. For now, simply creating a `Membership` object will suffice.

I ran into another issue when I tried to audit the `CustomUser` object. The audit library struggles with datetime serialization. I'll look into it. I have an idea where the issue occurs

### CODE REVIEW

File: apps/experiments/model_audit_fields.py [hunk 1]
```diff
@@ -33,3 +33,6 @@
     "confirmation_text",
     "team",
 ]
+
+TEAM_FIELDS = ["name", "slug", "members"]
+MEMBERSHIP_FIELDS = ["team", "user", "role"]
```
Comment (snopoke) [inline apps/experiments/model_audit_fields.py:40]: A bit odd to have these in the experiments app. Maybe move them to the `teams` app?

File: apps/experiments/model_audit_fields.py [hunk 2]
```diff
@@ -49,3 +51,6 @@
 MESSAGING_PROVIDER_FIELDS = ["type", "name", "team"]
 VOICE_PROVIDER_FIELDS = ["type", "name", "team"]
 LLM_PROVIDER_FIELDS = ["team", "type", "name", "llm_models"]
+
+# The auditing library struggles with dates. Let's ignore them for now
+CUSTOM_USER_FIELDS = [f.attname for f in AbstractUser._meta.fields if f.attname not in ["last_login", "date_joined"]]
```
Comment (snopoke) [inline apps/experiments/model_audit_fields.py:56]: Probably no need to audit these fields anyway.

**Review by snopoke — APPROVED** [review]

File: apps/experiments/model_audit_fields.py (hunk 1 above)
Comment (SmittieC) [reply apps/experiments/model_audit_fields.py:40]: Lol, fair

Comment (SmittieC) [reply apps/experiments/model_audit_fields.py]: https://github.com/dimagi/open-chat-studio/pull/215/commits/6ed76de0e0efc6cd0806de2429245e535b91fa5d

**Pull request merged**